File: src/main/java/main/commands/commands/misc/poll/PollEvents.java
package main.commands.commands.misc.poll;

import main.utils.GeneralUtils;
import net.dv8tion.jda.api.events.message.react.MessageReactionAddEvent;
import net.dv8tion.jda.api.events.message.react.MessageReactionRemoveEvent;
import net.dv8tion.jda.api.hooks.ListenerAdapter;
import org.jetbrains.annotations.NotNull;

import java.util.HashMap;

public class PollEvents extends ListenerAdapter {
    private final HashMap<Long, HashMap<Integer, Integer>> polls = PollCommand.getPollCache();


    @Override
    public void onMessageReactionAdd(@NotNull MessageReactionAddEvent event) {
        if (!polls.containsKey(event.getMessageIdLong())) return;
        if (event.getUser().isBot()) return;

        final var emoji = event.getReactionEmote().getEmoji();
        var countMap = polls.get(event.getMessageIdLong());
        var oldVal = countMap.get(GeneralUtils.parseNumEmoji(emoji));

        countMap.put(GeneralUtils.parseNumEmoji(emoji), ++oldVal);
    }

    @Override
    public void onMessageReactionRemove(@NotNull MessageReactionRemoveEvent event) {
        if (!polls.containsKey(event.getMessageIdLong())) return;
        if (event.getUser().isBot()) return;

        final var emoji = event.getReactionEmote().getEmoji();
        var countMap = polls.get(event.getMessageIdLong());
        var oldVal = countMap.get(GeneralUtils.parseNumEmoji(emoji));

        countMap.put(GeneralUtils.parseNumEmoji(emoji), --oldVal);
    }
}

File: src/main/kotlin/main/commands/slashcommands/misc/polls/PollEvents.kt
package main.commands.slashcommands.misc.polls

import dev.minn.jda.ktx.util.SLF4J
import main.commands.slashcommands.misc.polls.PollCommand.Companion.pollCache
import main.events.AbstractEventController
import main.utils.GeneralUtils
import net.dv8tion.jda.api.events.message.react.MessageReactionAddEvent
import net.dv8tion.jda.api.events.message.react.MessageReactionRemoveEvent

class PollEvents : AbstractEventController() {
    companion object {
        private val logger by SLF4J
    }

    override fun onMessageReactionAdd(event: MessageReactionAddEvent) {
        if (!pollCache.containsKey(event.messageIdLong)) return
        event.retrieveUser().queue { user ->
            if (user.isBot) return@queue

            val emoji = event.reaction.emoji.name
            val countMap = pollCache[event.messageIdLong]!!
            var oldVal = countMap[GeneralUtils.parseNumEmoji(emoji) - 1]!!
            countMap[GeneralUtils.parseNumEmoji(emoji) - 1] = ++oldVal
        }
    }

    override fun onMessageReactionRemove(event: MessageReactionRemoveEvent) {
        if (!pollCache.containsKey(event.messageIdLong)) return
        event.retrieveUser().queue { user ->
            if (user.isBot) return@queue

            val emoji = event.reaction.emoji.name
            val countMap = pollCache[event.messageIdLong]!!
            var oldVal = countMap[GeneralUtils.parseNumEmoji(emoji)]!!
            countMap[GeneralUtils.parseNumEmoji(emoji)] = --oldVal
        }
    }
}